src/mam4xx/wet_dep.hpp

```diff
@@ -1157,12 +1160,336 @@ void clddiag(const int nlev, const Real *temperature, const Real *pmid,
+                aero_model::lptr_dust_a_amode(imode))
+              f_act_conv[k] = f_act_conv_coarse_dust[k];
+            else if (aero_model::lmassptr_amode(lspec, imode) ==
+                     aero_model::lptr_nacl_a_amode(imode))
+              f_act_conv[k] = f_act_conv_coarse_nacl[k];
+          }
+        }
+        const int k_p1 = static_cast<int>(haero::min(k + 1, nlev - 1));
+        // OK, this is from the old mam4: Phase 2 is before Phase 1.
+        // Note that the phase loops goes from 2 to 1 in reverse order
+        // and the qqcw_sav is set first in phase 2 the used in phase 1.
+        if (lphase == 1) {
+          // traces reflects changes from modal_aero_calcsize and is the
+          // "most current" q
+          compute_q_tendencies_phase_1(
+              // These are the output values
+              scavt[k], bcscavt[k], rcscavt[k], rtscavt_sv.data(),
+              // The rest of the values are input only.
+              f_act_conv[k], scavcoefnum[k], scavcoefvol[k], totcond[k],
+              cmfdqr[k], conicw[k], evapc[k], evapr[k], prain[k], dlf[k],
+              cldt[k], cldcu[k], cldvst[k], cldvst[k_p1], cldvcu[k],
+              cldvcu[k_p1], sol_facti[k], sol_factic[k], sol_factb[k],
+              state_q(k, mm), ptend_q(k, mm), qqcw_sav(k, lspec, mm), pdel[k],
+              dt, mam_prevap_resusp_optcc, jnv, mm);
+
+        } else { // if (lphase == 2)
+          // There is no cloud-borne aerosol water in the model, so this
+          // code block should NEVER execute for lspec =
+          // nspec_amode(m)+1 (i.e., jnummaswtr = 2). The code only
+          // worked because the "do lspec" loop cycles when lspec =
+          // nspec_amode(m)+1, but that does not make the code correct.
+          compute_q_tendencies_phase_2(
+              // These are the output values
+              scavt[k], bcscavt[k], rcscavt[k], rtscavt_sv.data(),
+              qqcw_sav(k, lspec, mm),
+              // The rest of the values are input only.
+              progs, f_act_conv[k], scavcoefnum[k], scavcoefvol[k], totcond[k],
+              cmfdqr[k], conicw[k], evapc[k], evapr[k], prain[k], dlf[k],
+              cldt[k], cldcu[k], cldvst[k], cldvst[k_p1], cldvcu[k],
+              cldvcu[k_p1], sol_facti[k], sol_factic[k], sol_factb[k], pdel[k],
+              dt, mam_prevap_resusp_optcc, jnv, mm, k);
+        }
+      });
+}
+
+KOKKOS_INLINE_FUNCTION
+void update_q_tendencies(const ThreadTeam &team,
+                         Diagnostics::ColumnTracerView ptend_q,
+                         Kokkos::View<Real *> scavt, const int mm,
+                         const int nlev) {
+  Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, nlev),
+      KOKKOS_LAMBDA(int k) { ptend_q(k, mm) += scavt[k]; });
+}
 } // namespace wetdep
 
 /// @class WedDeposition
 /// Wet Deposition process for MAM4 aerosol model.
 class WetDeposition {
 public:
+  static constexpr int gas_pcnst = 40;
```

Comment:
We should probably declare `40` in aero config so that all processes can share this number.